influxdb3_server/src/query_executor.rs

```diff
@@ -574,3 +574,193 @@ impl<B: WriteBuffer> TableProvider for QueryTable<B> {
         provider.scan(ctx, projection, &filters, limit).await
     }
 }
+#[cfg(test)]
+mod tests {
+    use std::{num::NonZeroUsize, sync::Arc, time::Duration};
+
+    use arrow::array::RecordBatch;
+    use data_types::NamespaceName;
+    use datafusion::{assert_batches_sorted_eq, error::DataFusionError};
+    use futures::TryStreamExt;
+    use influxdb3_wal::WalConfig;
+    use influxdb3_write::{
+        persister::PersisterImpl, write_buffer::WriteBufferImpl, Bufferer, Level0Duration,
+    };
+    use iox_query::exec::{DedicatedExecutor, Executor, ExecutorConfig};
+    use iox_time::{MockProvider, Time};
+    use metric::Registry;
+    use object_store::{local::LocalFileSystem, ObjectStore};
+    use parquet_file::storage::{ParquetStorage, StorageId};
+
+    use crate::{
+        query_executor::QueryExecutorImpl, system_tables::table_name_predicate_error, QueryExecutor,
+    };
+
+    fn make_exec(object_store: Arc<dyn ObjectStore>) -> Arc<Executor> {
+        let metrics = Arc::new(metric::Registry::default());
+
+        let parquet_store = ParquetStorage::new(
+            Arc::clone(&object_store),
+            StorageId::from("test_exec_storage"),
+        );
+        Arc::new(Executor::new_with_config_and_executor(
+            ExecutorConfig {
+                target_query_partitions: NonZeroUsize::new(1).unwrap(),
+                object_stores: [&parquet_store]
+                    .into_iter()
+                    .map(|store| (store.id(), Arc::clone(store.object_store())))
+                    .collect(),
+                metric_registry: Arc::clone(&metrics),
+                // Default to 1gb
+                mem_pool_size: 1024 * 1024 * 1024, // 1024 (b/kb) * 1024 (kb/mb) * 1024 (mb/gb)
+            },
+            DedicatedExecutor::new_testing(),
+        ))
+    }
+
+    type TestWriteBuffer = WriteBufferImpl<MockProvider>;
+
+    async fn setup() -> (
+        Arc<TestWriteBuffer>,
+        QueryExecutorImpl<TestWriteBuffer>,
+        Arc<MockProvider>,
+    ) {
+        // Set up QueryExecutor
+        let object_store =
+            Arc::new(LocalFileSystem::new_with_prefix(test_helpers::tmp_dir().unwrap()).unwrap());
+        let persister = Arc::new(PersisterImpl::new(Arc::clone(&object_store) as _));
+        let time_provider = Arc::new(MockProvider::new(Time::from_timestamp_nanos(0)));
+        let executor = make_exec(object_store);
+        let level_0_duration = Level0Duration::new_5m();
+        let write_buffer = Arc::new(
+            WriteBufferImpl::new(
+                Arc::clone(&persister),
+                Arc::clone(&time_provider),
+                level_0_duration,
+                Arc::clone(&executor),
+                WalConfig {
+                    level_0_duration: Duration::from_millis(100),
```

Comment:
I'm about to change this in a follow on PR to use the Level0Duration type, which is only valid for 1m, 5m, or 10m durations. Can you update this test to use one of those?

Reply:
This is addressed in https://github.com/influxdata/influxdb/pull/25225/commits/ff6e9a41415bbf59aca312722d89a756e8bb79ce by using a 60 second duration. So switching to a `Level0Duration::ZERO` should be a straightforward change.